File: The Deck/Models/entity/RequestTable.go
package entity

import "time"

type RequestTable struct {
	Id          uint      `json:"id"`
	Description string    `json:"description" gorm:"type:text" validate:"required"`
	Status      string    `json:"status" gorm:"type:varchar(20)" validate:"required"`
	StartDate   time.Time `json:"start_date" gorm:"type:varchar(100)" validate:"required"`
	EndDate     time.Time `json:"end_date" gorm:"type:varchar(100)" validate:"required"`
	CreatedAt   time.Time `json:"created_at" gorm:"autoCreateTime" db:"created_at"`
	UpdatedAt   time.Time `json:"updated_at" gorm:"autoUpdateTime" db:"updated_at"`
	TableID     uint      `json:"table_id" gorm:"index"`
	CustomerID  uint      `json:"customer_id" gorm:"index"`
	Table       Table     `json:"table" gorm:"foreignKey:TableID"`
	Customer    Customer  `json:"customer" gorm:"foreignKey:CustomerID"`
}

File: frontend/the_deck/lib/Models/RequestTable.dart
import 'package:the_deck/Models/Table.dart';

class RequestTable {
  final int id;
  final int tableId;
  final int status;
  final String notes;
  Table? table;

  RequestTable({
    required this.id,
    required this.tableId,
    required this.status,
    required this.notes,
    this.table,
  });

  factory RequestTable.fromJson(Map<String, dynamic> json) {
    return RequestTable(
      id: json['id'] ?? 0,
      tableId: json['table_id'] ?? 0,
      status: json['status'] ?? 0,
      notes: json['notes'] ?? '',
    );
  }
}
